nanomq_cli/dds2mqtt/mqtt_client.c

```diff
@@ -368,13 +394,22 @@ mqtt_loop(void *arg)
 			dds_handlers->free(ddsmsg, DDS_FREE_ALL);
 			mqttmsg.payload = cJSON_PrintUnformatted(json);
 			mqttmsg.len     = strlen(mqttmsg.payload);
-			cJSON_free(json);
-
-			mqtt_publish(cli, cli->mqttsend_topic, 0,
+			cJSON_Delete(json);
+
+			dds_gateway_topic *dt = find_mqtt_topic(conf, hd->topic);
+			if (!dt) {
```

Comment:
Maybe this is better? To reduce the duplicated code
```
if (dt) {
....
}
free(hd->topic);
free(hd);
break;
```

Reply:
Okey. I will fix it.